src/api/routers/gcp/chat.py

```diff
@@ -84,38 +53,52 @@ def _parse_system_prompts(openai_messages) -> str:
         return system_prompts
 
 
-def to_vertex_anthropic(openai_messages):
-    message = []
+def to_vertex_anthropic(openai_messages, streaming=False):
+    """Convert OpenAI messages to Vertex Anthropic format."""
+    messages = []
     for m in openai_messages["messages"]:
         if m["role"] == "system":
             continue
-        message.append({"role": m["role"], "content": [{"type": "text", "text": m["content"]}]})
+
+        if isinstance(m["content"], str):
+            content = {"type": "text", "text": m["content"]} if streaming else [{"type": "text", "text": m["content"]}]
+        else:
+            content = m["content"] if streaming else [m["content"]]
```

Comment:
Could we do something like this?
```suggestion
        text = m["content"] if streaming else [m["content"]]
        if isinstance(m["content"], str):
            content = {"type": "text", "text": text}
        else:
            content = text
```
Definitely still feels a bit convoluted, but at least the number of branches are minimized.

I don't see any tests that cover this `streaming=True` case. Let's add some to make sure this behaves as expected.